h/views/activity.py

```diff
@@ -92,31 +95,76 @@ def group_search(request):
 
     return result
 
+
+@view_config(route_name='activity.user_search',
+             request_method='GET',
+             renderer='h:templates/activity/search.html.jinja2')
+def user_search(request):
+    if not request.feature('search_page'):
+        raise httpexceptions.HTTPNotFound()
+
+    username = request.matchdict['username']
+
+    result = search(request)
+
+    result['opts'] = {'search_username': username}
+    result['more_info'] = 'more_info' in request.params
+
+    user = request.find_service(name='user').fetch(
+        'acct:{username}@{authority}'.format(username=username,
+                                             authority=request.auth_domain))
```

Comment:
Hmm, the user service also caches user objects by userid, and of course if you only have the username and authority (and need the model object in order to get the userid) then you can't look up the model object in the cache by userid. So we may need a two-keyed cache that can look up objects by userid or bu (username, authority). (This definitely seems like something for a separate PR)